Jenkinsfile

```diff
@@ -1,12 +1,92 @@
 #!/usr/bin/env groovy
 
+def dockerRegistry = "329802642264.dkr.ecr.eu-west-1.amazonaws.com"
+def nodeImageVersion = "0.0.5"
+def nodeImage = "${dockerRegistry}/bbc-news/node-8-lts:${nodeImageVersion}"
+def nodeName
+def stageName = ""
+def getCommitInfo = {
+  infraGitCommitAuthor = sh(returnStdout: true, script: "git --no-pager show -s --format='%an' ${GIT_COMMIT}").trim()
+  appGitCommit = sh(returnStdout: true, script: "cd ${APP_DIRECTORY}; git rev-parse HEAD")
+  appGitCommitAuthor = sh(returnStdout: true, script: "cd ${APP_DIRECTORY}; git --no-pager show -s --format='%an' ${appGitCommit}").trim()
+  appGitCommitMessage = sh(returnStdout: true, script: "cd ${APP_DIRECTORY}; git log -1 --pretty=%B").trim()
+}
+
 pipeline {
   agent any
   options {
     timeout(time: 60, unit: 'MINUTES')
     timestamps ()
   }
+  environment {
+    APP_DIRECTORY = "app"
+    CI = true
+  }
   stages {
+    stage('Checkout application repo') {
+      when {
+        expression { env.BRANCH_NAME != 'latest' }
+      }
+      agent {
+        docker {
+          image "${nodeImage}"
+          args '-u root -v /etc/pki:/certs'
+        }
+      }
+      steps {
+        sh "rm -rf ${env.APP_DIRECTORY}"
+        checkout([
+          $class: 'GitSCM', 
+          branches: [[name: "*/${env.BRANCH_NAME}"]],
+          doGenerateSubmoduleConfigurations: false, 
+          extensions: [[
+            $class: 'RelativeTargetDirectory', 
+            relativeTargetDir: "${env.APP_DIRECTORY}"
+          ]], 
+          submoduleCfg: [], 
+          userRemoteConfigs: [[
+            credentialsId: 'github',
+            name: "origin/${env.BRANCH_NAME}",
+            url: 'https://github.com/bbc/simorgh.git'
+          ]]
+        ])
+        script {
+          getCommitInfo()
+          nodeName = "${env.node_name}".toString()
+        }
+      }
+      post {
+        always {
+          script {
+            stageName = env.STAGE_NAME
+          }
+        }
+      }
+    }
+    stage ('Run application tests') {
+      when {
+        expression { env.BRANCH_NAME != 'latest' }
+      }
+      agent {
+        docker {
+          image "${nodeImage}"
+          label nodeName
+          args '-u root -v /etc/pki:/certs'
+        }
+      }
+      steps {
+        sh 'make install'
+        sh 'make developmentTests'
+        sh 'make productionTests'
```

Comment:
The reason I have not parallelised `make developmentTests` & `make productionTests` is that this would involve stashing all files subsequent to the `make install` and then subsequently load them in a new instance of the node docker image. This would take more time than just running these tests serially.

Reply:
We should look into this problem, but as a separate piece of work. I believe there may be some optimisations we can make in order to benefit from executing stages in parallel.

Reply:
@BogdanDogaru Agreed.